### ISSUE TITLE
Getting stuck in Tap to Pay Screen when using auto-reconnect

### ISSUE THREAD

**Issue by ronzohan** [stripe/stripe-terminal-ios#338]
## Summary
I'm encountering a bug right now that happens when you try to collect payment and then try to background the app. In most occasions, the _Tap to Pay_ screen will just Cancel automatically. But sometimes, the _Tap to Pay_ screen will show up but it will get stuck. 

The `TapToPayConnectionConfigurationBuilder` is currently set to auto-reconnect on reader disconnection in the setup above.

## Code to reproduce

Here is a video showing me trying to reproduce it (at around 00:10 where the _Tap to Pay_ screen gets stuck). https://drive.google.com/file/d/1-FHLkrYrhhFiOqWqUhqAo3PqWlgsEmnV/view?usp=sharing

## iOS version
iOS 18.1.1

## Installation method
SPM

## SDK version
4.1.0

**Comment by maggiewhite-stripe** [comment]
Thanks for reporting this issue @ronzohan. This issue is most likely to be in Apple's logic, and they find sysdiagnose logs to be the most helpful. Can you collect a sysdiagnose log after this issue happens and share it with Stripe's support team? Here are more details about how to collect a sysdiagnose log:
https://it-training.apple.com/tutorials/support/sup075/

**Comment by maggiewhite-stripe** [comment]
@ronzohan Thanks for reporting this issue. I noticed that in the video, you are backgrounding the application immediately after pushing the "pay" button. Does this issue happen if you background the application after the payment screen is already being displayed? Can you share a payment intent when this issue is happening so we can find the logs?

**Comment by yliu342** [comment]
>Does this issue happen if you background the application after the payment screen is already being displayed? 

No, if the payment screen is shown, then when to try to background the app it will simply dismiss the tap to pay payment screen first.

>Can you share a payment intent when this issue is happening so we can find the logs?

Unfortunately Ron is on holiday until the 10th of Jan. I will try to replicate on my device and if I can replicate then I will reply here with an payment intent id

**Comment by yliu342** [comment]
Hi @maggiewhite-stripe I did reproduce it once `pi_3QXC8EEhqOG9KqpJ0Jgzx2HR` this is the payment intent id

**Comment by yliu342** [comment]
I think I found a relative reliable way to reproduce:
- press a button to trigger the payment screen. (payment screen will take a small amount time to appear)
- Then quickly background the app (at this stage payment screen is still preparing)
- On the home screen, try to wipe the home bar which will try to switch app, then if you do it in the right time(i.e. the same time the payment screen will be presented) Then you will see the app get stuck for some time like 10-20 seconds.

**Comment by maggiewhite-stripe** [comment]
@yliu342 Thank you! I was able to pull up the logs on our end from `pi_3QXC8EEhqOG9KqpJ0Jgzx2HR`. The UX shown here is owned by Apple's logic, so I've reported this issue to them and will let you know if there are any updates I can share.

**Comment by maggiewhite-stripe** [comment]
@yliu342 Apple has asked if you can provide a sysdiagnose (iOS system log file) to help them debug. The instructions on how to collect a sysdiagnose are here:
https://it-training.apple.com/tutorials/support/sup075/

**Comment by ronzohan** [comment]
Hi @maggiewhite-stripe, if it helps, I've also been able to reproduce it again with this payment intent `pi_3QgvbFEhqOG9KqpJ0DkNUKeO`.

I've also uploaded the sysdiagnose in this [Google Drive link](https://drive.google.com/file/d/1L_kT__JUWmfOTA95W1jgBugU07rpmOIT/view?usp=sharing) since it was quite large (~300mb) to be attached here in GitHub

**Comment by maggiewhite-stripe** [comment]
@ronzohan Thank you for the quick response! I shared the sysdiagnose with Apple

**Comment by henryx-stripe** [comment]
Hi @ronzohan could you verify that the Tap To Pay profile [0] has been installed before capturing the logs? Apple is saying that the logs are redacted because the profile was not installed. Thanks in advance!

[0] https://developer.apple.com/bug-reporting/profiles-and-logs/?name=tap%20to%20pa

**Comment by ronzohan** [comment]
Hi @henryx-stripe, I've attached my sysdiagnose to the link below for a device with this issue. Before I capture the logs, it should also have the Tap to Pay profile installed.

https://drive.google.com/file/d/13Ou5Wiy7XIMAf2XcA298Jth76vxWfqVO/view?usp=sharing

**Comment by henryx-stripe** [comment]
Thank you! I have forwarded the logs to apple. I will update here as soon as I get some more information from them.

**Comment by maggiewhite-stripe** [comment]
Apple has identified the issue and is working on a fix - we will update this ticket again when the fix is available.
